ocs_ci/framework/__init__.py

```diff
@@ -130,9 +133,9 @@ class MultiClusterConfig:
     # multiple cluster contexts
     def __init__(self):
         # Holds all cluster's Config() object
+        self.thread_local_data = local()
```

Comment:
this way we create multiple threading.local objects of MultiClusterConfig. Is it the desired behavior? 
Maybe we want isolate MultiClusterConfig from multiple creations? Make it Singleton?

Reply:
Yes it's good candidate to define MultiClusterConfig as singleton but it's not a part of this POC and PR to do refactoring of current implementation from this point of view.

In whole framework we have only one instance of config object defined here:

https://github.com/red-hat-storage/ocs-ci/blob/c4b295570eddd7a0e61231738c8adbf2a2fd11ef/ocs_ci/framework/__init__.py#L503

And we are everywhere referring to the same object - so we don't actually doing in whole framework the other instance of this class.